examples/ota-provider-app/ota-provider-common/BdxOtaSender.cpp

```diff
@@ -88,6 +76,13 @@ void BdxOtaSender::HandleTransferSessionOutput(TransferSession::OutputEvent & ev
         acceptData.Length       = mTransfer.GetTransferLength();
         VerifyOrReturn(mTransfer.AcceptTransfer(acceptData) == CHIP_NO_ERROR,
                        ChipLogError(BDX, "%s: %s", __FUNCTION__, chip::ErrorStr(err)));
+
+        // Store the file designator used during block query
+        uint16_t fdl       = 0;
+        const uint8_t * fd = mTransfer.GetFileDesignator(fdl);
+        memcpy(mFileDesignator, fd, fdl);
+        mFileDesignator[fdl] = 0;
```

Comment:
Should we do a bounds check here on fdl to ensure it will not cause out of bounds access on mFileDesignator?